### PR TITLE
[Merged by Bors] - Cleanup `boa_tester`

### PR DESCRIPTION
**Pull request by jedel1043**
Just some quality changes to improve the maintainability of the tester:

- Replaces `anyhow` with `color_eyre` to have a better output on errors/panics.
- Changes the ignore file to a TOML file and replaces all parsing logic with the `toml` crate.
- Adds a `ignored` field on all `Test`s to simplify run logic.
- Replaces the global `IGNORED` with an `ignored` argument on the CLI.

### CODE REVIEW

**Review by Razican — APPROVED** [review]
Looks good to me :) thanks for the changes!!

File: boa_tester/src/read.rs [hunk 1]
```diff
@@ -138,38 +124,60 @@ pub(super) fn read_harness(test262_path: &Path) -> anyhow::Result<Harness> {
...
 
     let mut suites = Vec::new();
     let mut tests = Vec::new();
 
+    // !ignored.contains_any_flag(self.flags)
+    // && !ignored.contains_test(&self.name)
+    // && !self
+    //     .features
+    //     .iter()
+    //     .any(|feat| ignored.contains_feature(feat))
```
Comment (raskad) [inline boa_tester/src/read.rs:148]: Some remaining commented out code.

**Review by raskad — APPROVED** [review]
Nice work!

**Pull request closed without merging**